torchgeo/datasets/oscd.py

```diff
@@ -240,7 +242,7 @@ def _load_target(self, path: Path) -> Tensor:
             array: np.typing.NDArray[np.int_] = np.array(img.convert('L'))
             tensor = torch.from_numpy(array)
             tensor = torch.clamp(tensor, min=0, max=1)
-            tensor = tensor.to(torch.long)
+            tensor = tensor.to(torch.float)
```

Comment:
Why would the target be a float?

Reply:
The loss function `BCEWithLogitsLoss` expects the target to be a float.

Reply:
Why do we have to use BCEWithLogitsLoss? Can we use BCELoss instead?